examples/gateway-reshard.rs

```diff
@@ -61,73 +52,50 @@ async fn gateway_runner(client: Arc<Client>, mut shards: Vec<Shard>) {
 
                 continue;
             }
-            None => break,
         };
 
-        tokio::spawn(event_handler(Arc::clone(&client), event));
-    }
-}
-
-async fn event_handler(client: Arc<Client>, event: Event) -> anyhow::Result<()> {
-    match event {
-        Event::MessageCreate(message) if message.content == "!ping" => {
-            client
-                .create_message(message.channel_id)
-                .content("Pong!")?
-                .await?;
-        }
-        _ => {}
+        tracing::debug!(?event, shard = ?shard.id(), "received event");
     }
-
-    Ok(())
 }
 
-// Instrument to differentiate between the logs produced here and
-// in `gateway_runner`.
+// Instrument to differentiate between the logs produced here and in `runner`.
 #[tracing::instrument(skip_all)]
 async fn reshard(
     client: &Client,
     config: Config,
     config_callback: impl Fn(ShardId, ConfigBuilder) -> Config,
-) -> anyhow::Result<Option<Vec<Shard>>> {
+) -> anyhow::Result<Vec<Shard>> {
+    // Reshard every eight hours. This is an arbitrary number.
     const RESHARD_DURATION: Duration = Duration::from_secs(60 * 60 * 8);
 
-    // Reshard every eight hours.
     time::sleep(RESHARD_DURATION).await;
 
     let mut shards = stream::create_recommended(client, config, config_callback)
         .await?
         .collect::<Vec<_>>();
 
-    let mut identified = vec![false; shards.len()];
-    // Don't deserialize any events (with `ShardEventStream`) as the already
-    // running shards will handle them (the events are duplicated).
+    // Before swapping the old and new list of shards, try to identify them.
+    // Don't try too hard, however, as large bots may never have all shards
+    // identified at the same time.
+    let mut identified = vec![0; shards.len()];
```

Comment:
I would prefer keeping the booleans here since they carry over more meaning in this context than an integer that will be 0 or 1. Sure that means we can't use `Iterator::sum`, but it should convey the intent behind this better.